src/node.cc

```diff
@@ -298,14 +298,21 @@ MaybeLocal<Value> Environment::BootstrapInternalLoaders() {
     return MaybeLocal<Value>();
   }
   CHECK(loader_exports->IsObject());
```

Comment:
You should remove this check, otherwise there's no point to the changes below.

That said, `loader_exports->IsObject() == false` means a grave bug. Therefore, it's good to have the check. Therefore, the changes below don't really make sense.

Reply:
Will the `IsObject` check return false if it's empty? The v8 docs doesn't explicitly say this - https://v8docs.nodesource.com/node-0.8/dc/d0a/classv8_1_1_value.html#afbcb69694b71bd31e979423cf499a527

If not, then my changes are relevant.

Reply:
@nodejs/v8

Reply:
It'll segfault. An empty handle is basically a nullptr.

Reply:
> It'll segfault. An empty handle is basically a nullptr.

There's no actual guarantee a nullptr deref results in a segfault. It results in undefined behaviour, of which a segfault is a perfectly valid result.

Reply:
Hey, a language lawyer. You're correct if you base yourself on the C++ spec but we target a list of systems where nullptr == segfault.

Reply:
You're *assuming* that it gets compiled to an actual deref, rather than dereferences causing later null checks to be optimised out.

Reply:
You're *assuming* that it gets compiled to an actual deref, rather the UB wreaking havoc / causing later null checks to be optimised out.